sinks/signalfx/signalfx.go

```diff
@@ -430,17 +430,32 @@ METRICLOOP: // Convenience label so that inner nested loops and `continue` easil
 				dims[key] = kv[1]
 			}
 		}
-		// Copy common dimensions
-		for k, v := range sfx.commonDimensions {
-			dims[k] = v
-		}
+
+		// metric-specified API key, if present, should override the common dimension
 		metricKey := ""
+		overridePresent := false
+
 		if sfx.varyBy != "" {
 			if val, ok := dims[sfx.varyBy]; ok {
 				metricKey = val
+				overridePresent = true
+			} else {
+				if val, ok := sfx.commonDimensions[sfx.varyBy]; ok {
+					metricKey = val
```

Comment:
Do we need this `else` at all? Because all common dimensions are copied into `dims` anyways on L451.

Idea: maybe we get rid of this `else` block and rename `overridePresent` to `metricVaryByOverridePresent`

Reply:
Ah yeah, I reordered this entire secion from a previous version, so this is no longer needed anymore.